### PR TITLE
add getArticleById and tests

### PR DESCRIPTION
**Pull request by jss118**
tested the /api/articles/:article_id endpoint..

also made a separate test for all the 404 errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -8,7 +8,18 @@ afterAll(() => db.end());
...
 beforeEach(() => seed(testData));
 
-describe("/api/topics", () => {
+describe("ALL /*", () => {
+  test("Status: 404 with an error message for a none-existing endpoint", () => {
+    return request(app)
+      .get("/api/invalid-endpoint")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Sorry, request invalid..");
```
Comment (Rosem10) [inline __tests__/app.test.js:17]: On a personal note, I think it's THEM who should be sorry, not you - but the code is good.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -23,12 +34,21 @@ describe("/api/topics", () => {
...
+    const { body } = await request(app).get("/api/articles/1").expect(200);
+    expect(typeof body).toBe("object");
+    expect(body.author).toEqual(expect.any(String));
+    expect(body.title).toEqual(expect.any(String));
+    expect(body.article_id).toEqual(expect.any(Number));
+    expect(body.topic).toEqual(expect.any(String));
+    expect(body.created_at).toBe("2020-07-09T20:11:00.000Z");
+    expect(body.votes).toEqual(expect.any(Number));
+  });
+  test("Status: 400 recieves an error when a bad request is made", async () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:50]: What is the request that is made is of the right type, but refers to an article not present in the database (e.g id 9999 when we know there's only a few in the test db)?

**Review by Rosem10 — CHANGES_REQUESTED** [review]
One more little bit of error handling please!

**Review by HeatherEver — APPROVED** [review]
nice work! Go ahead and merge

**Pull request merged**